### PR TITLE
Topbar

### PR DESCRIPTION
**Pull request by oliferwolox**
## Summary

Add topbar with Logout button and links to Matches screen.

## Screenshots

topbar-matches2 0

## Trello Card

https://trello.com/c/NzYSkILF/24-routing-ii-topbar
https://trello.com/c/NQ9zrCXN/7-react-app-ii-other-views

### CODE REVIEW

File: react/tic-tac-toe/src/app/screens/Game/index.js [hunk 1]
```diff
@@ -55,19 +55,21 @@ class Game extends Component {
...
-          onClick={(i) => this.handleClick(i)}
-         />
-      </div>
-      <div className={styles.gameInfo}>
-        <div>{status}</div>
-        <ol>{moves}</ol>
-      </div>
-      <TableMatches />
-    </div>
+      <>
```
Comment (carlosbollero) [inline react/tic-tac-toe/src/app/screens/Game/index.js]: Just to know, this isn't supported by some tools versions.
https://reactjs.org/blog/2017/11/28/react-v16.2.0-fragment-support.html#support-for-fragment-syntax

File: react/tic-tac-toe/src/app/screens/Matches/index.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import React, { Component } from 'react';
+import TableMatches from './components/TableMatches';
+import Topbar from '../../components/Topbar';
+
+class Matches extends Component {
```
Comment (carlosbollero) [inline react/tic-tac-toe/src/app/screens/Matches/index.js]: This could be a function.

File: react/tic-tac-toe/src/app/components/Topbar/index.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+import styles from './styles.module.scss';
+
+class Topbar extends Component {
+    goToMatches = () => (this.props.history.push("/matches"));
+
+    goToGame = () => (this.props.history.push("/game"));
+    
+    render() {
+        return (
+      <div className={styles.topbar}>
```
Comment (carlosbollero) [inline react/tic-tac-toe/src/app/components/Topbar/index.js]: I'm pretty sure we have an indentation problem here 😄

File: react/tic-tac-toe/src/app/index.js [hunk 4]
```diff
@@ -33,13 +35,14 @@ class App extends Component{
...
-  error: store.login.error
+  errorMessage: store.login.errorMessage
 })
 
 const mapDispatchToProps = dispatch => ({
   checkCredentials: ({email, password}) => dispatch(actionsCreators.postUser(email,password)),
   setAuthenticated: (boolean) => dispatch(actionsCreators.setAuthenticated(boolean)),
-  setToken: (token) => dispatch(actionsCreators.setToken(token))
+  setToken: (token) => dispatch(actionsCreators.setToken(token)),
+  logout: () => dispatch(actionsCreators.logout()),
```
Comment (LucasZibell) [inline react/tic-tac-toe/src/app/index.js]: Do you need this function here? Can you map it on the desired component? 🤔

File: react/tic-tac-toe/src/redux/login/actions.js [hunk 5]
```diff
@@ -40,7 +41,12 @@ const actionsCreators = {
   setToken: (token) => ({
     type: actions.SET_TOKEN,
     payload: token
-  })
+  }),
+  logout: () => {
+    delete api.headers['Authorization'];
```
Comment (LucasZibell) [inline react/tic-tac-toe/src/redux/login/actions.js]: I think you have a method to remove the header without using delete

File: react/tic-tac-toe/src/app/screens/Login/index.js [hunk 6]
```diff
@@ -11,18 +11,19 @@ class Login extends Component {
         <LoginForm
           onSubmit={this.submit}
           isLoading={this.props.loading}
-          hasError={this.props.hasError}
+          hasError={this.props.errorMessage=== '' ? false: true}
```
Comment (carlosbollero) [inline react/tic-tac-toe/src/app/screens/Login/index.js]: `hasError={!!this.props.errorMessage}`

**Pull request merged**